cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -185,15 +192,38 @@ where
 		};
 		(&*self.client).import_block(block_import_params, Default::default()).await?;
 
+		let mut roots = self
+			.runtime_api
+			.runtime_api()
+			.intermediate_roots(&BlockId::Number(block_number))?;
```

Comment:
I understand why you did it, but would be great to not store execution trace in state if there is a way, maybe add TODO somewhere?

Reply:
Sure, I can add a TODO. I did try to not introduce the executive pallet for collecting the intermediate storage roots in the state, let's say, try calculating the root in block builder by accessing to the state backend directly, something like that, but just failed, maybe there are some other ways I'm not aware of, I'm not sure.

Reply:
The idea would be to remove info from storage before finalizing block. The way to do that I can come up immediately is to emit an event, but there should be a more direct way as well I think, but we'll probably have to customize more stuff for that.

Reply:
Manipulating the state backend directly will bring no more storage, it's just to calculate the storage root after applying each extrinsic, so you don't have to remove anything that way. But yeah, it requires some deeper customization.

Reply:
You put intermediate state roots (trace) into the storage to be able to retrieve it here. But it is a derived information, just like events, that doesn't have to be stored all the time in the state tree. So it uses more storage than if we didn't store trace in the state.

Reply:
Yeah, the current way uses the state storage to store the intermediate storage roots, which is just a workaround. The point of my last comment describes the ideal approach in my mind which is to manipulate the state backend for calculating the storage root directly, but I failed to make it work right now.

Reply:
Sure, not a blocker, but TODO would be nice to have

Reply:
TODO added.